matchzoo/embedding.py

```diff
@@ -1,143 +1,68 @@
-        num_shared_words = 0
-
-        with open(self._embedding_file, 'rb') as embedding_file_ptr:
-            # Detect embedding_dim from first line
-            # TODO when embedding dim equals one, this will not work.
-            last_pos = embedding_file_ptr.tell()
-            first_line = embedding_file_ptr.readline()
-            if len(first_line.split(b" ")) == 2:
-                _, self._embedding_dim = map(int, first_line.split(b" "))
-            else:
-                entries = first_line.rstrip().split(b" ")
-                vec = entries[1:]
-                self._embedding_dim = len(vec)
-                embedding_file_ptr.seek(last_pos)
-
-            self._embedding_mat = np.random.uniform(-0.25, 0.25,
-                                                    (num_dict_words,
-                                                     self._embedding_dim))
-
-            # Go through all embedding file
-            for line in tqdm(embedding_file_ptr):
-                # Explicitly splitting on " " is important, so we don't
-                # get rid of Unicode non-breaking spaces in the vectors.
-                entries = line.rstrip().split(b" ")
-
-                word, vec = entries[0], entries[1:]
-
-                if self._embedding_dim != len(vec):
-                    raise RuntimeError(
-                        "Vector for token {} has {} dimensions, but "
-                        "previously read vectors have {} dimensions. "
-                        "All vectors must have the same number of "
-                        "dimensions.".format(word, len(vec),
-                                             self._embedding_dim))
-
-                try:
-                    if isinstance(word, bytes):
-                        word = word.decode('utf-8')
-                except Exception:
-                    logger.warning(
-                        "Skipping non-UTF8 token {}".format(repr(word)))
-                    continue
-
-                index = term_index.get(word, None)
-                if index:
-                    self._embedding_mat[index] = np.array(vec).astype(float)
-                    self._index_state[index] = 1
-                    num_shared_words += 1
-
-            # init tht OOV word embeddings
-            for word in term_index:
-                index = term_index[word]
-                if index not in self._index_state:
-                    self._index_state[index] = 2
-
-            self._index_state[0] = 0
-            self._embedding_mat[0] = np.zeros([self._embedding_dim])
+    def matrix(self):
+        """Getter."""
+        if not self._matrix:
+            raise ValueError('Please load embedding.')
+        return self._matrix
+
+    @matrix.setter
+    def matrix(self, matrix_instance):
+        """Setter."""
+        self._matrix = matrix_instance
+
+    @classmethod
+    def load(cls, dir: str, embedding_type: str = 'random'):
```

Comment:
`dir` is not needed for randomly initialized embeddings. I suggest separating this into to class methods, one to `load`, one to `randomly_initialize`. Also, it may be a good idea to take a random variable (e.g. scipy.rv_continueous) or an initializer (e.g. a lambda returns a random number) to initialize an embedding, but we can leave that for later.